source/CDC.DEX.FHIR.Function/CDC.DEX.FHIR.Function.ProcessExport/ProcessExport.cs

```diff
@@ -41,115 +44,152 @@ public ProcessExport(IHttpClientFactory httpClientFactory, IConfiguration config
+
+                    if (flagFhirResourceCreatedExportFunctionFlatten)
+                    {
+                        //flatten
+                        string flattenedJson = FlattenJsonResource(fhirResourceToProcessJObject);
+
+                        string pathToWrite = fhirResourceToProcessJObject["resourceType"].Value<string>();
+                        //get profile data for sorting bundles
+                        if (fhirResourceToProcessJObject["resourceType"].Value<string>() == "Bundle")
+                        {
+                            string profilePath = fhirResourceToProcessJObject["meta"]["profile"][0].Value<string>();
+                            profilePath = profilePath.Substring(profilePath.LastIndexOf("/"));
+                            pathToWrite += "/" + profilePath;
+                        }
+                        pathToWrite += "/" + fhirResourceToProcessJObject["id"].Value<string>();
+                        filesToWrite.Add(pathToWrite, flattenedJson.ToString());
+                    }
+                    else
+                    {
+                        string pathToWrite = fhirResourceToProcessJObject["resourceType"].Value<string>();
+                        //get profile data for sorting bundles
+                        if (fhirResourceToProcessJObject["resourceType"].Value<string>() == "Bundle")
+                        {
+                            string profilePath = fhirResourceToProcessJObject["meta"]["profile"][0].Value<string>();
+                            profilePath = profilePath.Substring(profilePath.LastIndexOf("/"));
+                            pathToWrite += "/" + profilePath;
                         }
+                        pathToWrite += "/" + fhirResourceToProcessJObject["id"].Value<string>();
+                        filesToWrite.Add(pathToWrite, fhirResourceToProcessJObject.ToString());
                     }
+                }
 
-                    // END GET FHIR RESOURCE SECTION
+                // FOR CONNECTATHON ALWAYS MAKE A FLATTEN VERSION, IN A SEPERATE DIRECTORY
+                string flattenedJsonTemp = FlattenJsonResource(fhirResourceToProcessJObject);
 
-                    // START WRITING TO DATA LAKE SECTION
+                string pathToWriteTemp = "Flatten/"+fhirResourceToProcessJObject["resourceType"].Value<string>();
+                //get profile data for sorting bundles
+                if (fhirResourceToProcessJObject["resourceType"].Value<string>() == "Bundle")
+                {
+                    string profilePath = fhirResourceToProcessJObject["meta"]["profile"][0].Value<string>();
+                    profilePath = profilePath.Substring(profilePath.LastIndexOf("/"));
+                    pathToWriteTemp += "/" + profilePath;
+                }
+                pathToWriteTemp += "/" + fhirResourceToProcessJObject["id"].Value<string>();
+                filesToWrite.Add(pathToWriteTemp, flattenedJsonTemp.ToString());
```

Comment:
@bennettn4 will be we writting the flattened file twice if the feature flag is on?

Reply:
currently, yes
we will need to rework the exact mechanics of writing to the datalake once the live data lake is available to us, it's looking like we will need to push data to it via an API but the datalake team hasn't finalized yet